src/bootupd.rs

```diff
@@ -93,6 +93,15 @@ pub(crate) fn install(
             continue;
         }
 
+        // skip components that don't have an update metadata
```

Comment:
Hmm this one is interesting. I think actually the problem is we're iterating over the static list of components, when actually I think we should just accept whatever the updated components in the updates set are to start with or so.

Although it raises a different point - it would probably make sense in the future to handle the case where a component is removed. (e.g. an OS update drops out BIOS support and we would then in theory clear out the BIOS partition e.g.)

And to do that, we do need to iterate over the components we know (as this is doing) and intersect them with the update set.